File: 프로그래머스/2/12923. 숫자 블록/숫자 블록.py
def solution(begin, end):
    answer = []
    
    for i in range(begin, end + 1):
        # 1이면 0, 0이면 1
        k = int(i!=1)
        max_num = 1
        for j in range(2, int(i**0.5)+1):
            if i%j == 0:
                if (i//j)<=10000000:
                    k=i//j
                    break
                else:
                   max_num  = j  
        if k == 1 :
            answer.append(max_num)
        else:
            answer.append(k)
        
    return answer

File: 프로그래머스/2/12923. 숫자 블록/숫자 블록.js
function solution(begin, end) {
    var answer = [];
    
    // num을 넣으면 약수중에서 자신을 빼고 가장 큰 수를 리턴
    function check(num){       
        var checkArr=[];        
        if(num===1){
            return 0;
        }
        // 약수 구하기
        for(var i =2;i<=Math.sqrt(num);i++){
            if(num%i===0){
                checkArr.push(i);
                if(num/i <= 1e7){
                    return num/i;
                }
            }
        }
        if(checkArr.length!==0){
            return checkArr[checkArr.length-1];
        }     
        // 없다면 1을 리턴 (1은 모두 나눠짐.)
        return 1;
    }
    
    
    for(var i = begin;i<=end;i++){
        var checkNum = check(i);
        if(checkNum!==undefined){
             answer.push(checkNum);
        }
    }
    
    return answer;
}